test-context/src/lib.rs

```diff
@@ -0,0 +1,214 @@
+use postgresql_embedded::{PostgreSQL, Settings, VersionReq};
+use std::env;
+use std::env::current_dir;
+use std::io::ErrorKind;
+use std::path::{Path, PathBuf};
+use test_context::AsyncTestContext;
+use tokio::io::AsyncReadExt;
+use tokio_util::bytes::Bytes;
+use tokio_util::io::ReaderStream;
+use tracing::{info_span, instrument, Instrument};
+use trustify_common as common;
+use trustify_module_ingestor::graph::Graph;
+use trustify_module_ingestor::model::IngestResult;
+use trustify_module_ingestor::service::{Format, IngestorService};
+use trustify_module_storage::service::fs::FileSystemBackend;
+
+#[allow(dead_code)]
+pub struct TrustifyContext {
+    pub db: common::db::Database,
+    pub graph: Graph,
+    pub storage: FileSystemBackend,
+    pub ingestor: IngestorService,
+    postgresql: Option<PostgreSQL>,
+}
+
+impl TrustifyContext {
+    pub async fn ingest_documents<'a, P: IntoIterator<Item = &'a str>>(
+        &self,
+        paths: P,
+    ) -> Result<Vec<IngestResult>, anyhow::Error> {
+        let workspace_root = find_workspace_root()?;
+        let test_data = workspace_root.join("etc").join("test-data");
+
+        let mut results = Vec::new();
+
+        for path in paths {
+            results.push(self.ingest_document_inner(&test_data, path).await?);
+        }
+
+        Ok(results)
```

Comment:
I always balk at the "iterate while pushing to another mutable container" pattern, but the futures complicate things. I came up with this, but I'm not sure I like it better:

```suggestion
        let futures = paths
            .into_iter()
            .map(|path| async { self.ingest_document_inner(&test_data, path).await });
        join_all(futures).await.into_iter().collect()
```